src/cohort_manager.js

```diff
@@ -0,0 +1,117 @@
+const Cohorts = require('./cohorts.js')
+
+class CohortManager {
+  constructor() {
+    this.cohortList = []
+    this.studentId = 1
+    this.max_capacity = 5
+  }
+
+  createCohort(cohortName) {
+    const cohortToAdd = new Cohorts(cohortName)
+    this.cohortList.push(cohortToAdd)
+    return this.cohortList
+  }
+
+  searchByCohortName(cohortName) {
+    let cohortFound = false
+    let cohort
+    for (let i = 0; i < this.cohortList.length; i++) {
+      if (this.cohortList[i].name === cohortName) {
+        cohortFound = true
+        cohort = this.cohortList[i]
+        break
+      }
+    }
+    if (cohortFound === false) {
+      return 'COHORT NOT FOUND'
+    }
+    return cohort
+  }
+
+  addStudent(cohortName, studentName, gitHub, email) {
+    for (let i = 0; i < this.cohortList.length; i++) {
+      if (this.cohortList[i].name === cohortName) {
+        const currentCohort = this.cohortList[i]
+        if (currentCohort.students.length < this.max_capacity) {
+          currentCohort.addStudentToStudentList(
+            this.studentId,
+            studentName,
+            gitHub,
+            email
+          )
+          this.studentId++
+          return currentCohort
+        } else {
+          return 'THE COHORT IS FULL'
+        }
+      }
+    }
+    return 'COHORT NOT FOUND'
+  }
+
+  removeCohort(cohortName) {
+    let cohortFound = false
+    for (let i = 0; i < this.cohortList.length; i++) {
+      if (this.cohortList[i].name === cohortName) {
+        this.cohortList.splice(i, 1)
+        cohortFound = true
+        return this.cohortList
+      }
+    }
+    if (cohortFound === false) {
+      return 'NOT A VALID COHORT NAME'
+    }
+  }
+
+  findCohort(cohort) {
+    let foundCohort = null
+    for (let i = 0; i < this.cohortList.length; i++) {
+      if (this.cohortList[i].name === cohort) {
+        foundCohort = this.cohortList[i]
+      }
+    }
+    return foundCohort
+  }
+
+  removeStudent(cohort, student) {
+    const theCohort = this.findCohort(cohort)
+    if (theCohort === null) {
+      return 'COHORT NOT FOUND'
+    } else {
+      for (let i = 0; i < theCohort.students.length; i++) {
+        if (theCohort.students[i].firstName === student) {
+          theCohort.students.splice(i, 1)
+          return theCohort
+        }
+      }
+    }
+  }
+
+  findCohortsWithStudents() {
+    const cohortsWithStudents = []
+    for (let i = 0; i < this.cohortList.length; i++) {
+      if (this.cohortList[i].students.length === 0) {
+        continue
+      }
+      if (this.cohortList[i].students.length !== 0) {
+        cohortsWithStudents.push(this.cohortList[i])
+      }
+    }
+    return cohortsWithStudents
+  }
+
+  searchByID(id) {
```

Comment:
<img width="420" alt="Screenshot 2022-04-04 at 09 56 33" src="https://user-images.githubusercontent.com/98287394/161509829-fec214a6-222d-4be2-b6bb-58cb9e73b85d.png">
This is how I tackled this section, it seems to be a lot simpler, did you find it easier to do it your way or is their extra functionality etc?